airflow/serialization/serialized_objects.py

```diff
@@ -626,3 +653,60 @@ def from_dict(cls, serialized_obj: dict) -> 'SerializedDAG':
         if ver != cls.SERIALIZER_VERSION:
             raise ValueError("Unsure how to deserialize version {!r}".format(ver))
         return cls.deserialize_dag(serialized_obj['dag'])
+
+
+class SerializedTaskGroup(TaskGroup, BaseSerialization):
+    """
+    A JSON serializable representation of TaskGroup.
+    """
+    @classmethod
+    def serialize_task_group(cls, task_group: TaskGroup) -> Union[Dict[str, Any], None]:
+        """Serializes TaskGroup into a JSON object.
+        """
+        if not task_group:
+            return None
+
+        serialize_group = {}
+        serialize_group["_group_id"] = task_group._group_id  # pylint: disable=protected-access
+
+        serialize_group['children'] = {  # type: ignore
+            label: (DAT.OP, child.task_id)
+            if isinstance(child, BaseOperator) else
+            (DAT.TASK_GROUP, SerializedTaskGroup.serialize_task_group(child))
+            for label, child in task_group.children.items()
+        }
+        serialize_group['tooltip'] = task_group.tooltip
+        serialize_group['ui_color'] = task_group.ui_color
+        serialize_group['ui_fgcolor'] = task_group.ui_fgcolor
```

Comment:
nitpick, looks like these individual dictionary set operation can be combined into the dict initialization statement?

```python
serialize_group = {
    "tooltip": task_group.toolip,
    ...
}
```